modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue

```diff
@@ -325,31 +335,71 @@ export default {
         draft: true
       });
     },
-    async getPieces () {
-      if (this.holdQueries) {
-        return;
-      }
-
-      this.holdQueries = true;
-
+    async requestPieces(page = 1, mergeOptions = {}) {
       const {
         currentPage, pages, results, choices
       } = await this.request({
+        ...mergeOptions,
         ...(
           this.moduleOptions.managerApiProjection &&
           { project: this.moduleOptions.managerApiProjection }
         ),
-        page: this.currentPage
+        page
       });
 
-      this.currentPage = currentPage;
-      this.totalPages = pages;
-      this.items = results;
-      this.filterChoices = choices;
-      this.holdQueries = false;
+      return {
+        currentPage,
+        pages,
+        results,
+        choices
+      };
     },
-    async getAllPiecesTotal () {
+    async requestAllPiecesTotal() {
       const { count: total } = await this.request({ count: 1 });
+      return total;
+    },
+    async requestData(currentPage = 1) {
+      const pieces = await this.requestPieces(currentPage);
+      const total = await this.requestAllPiecesTotal();
+
+      return {
+        ...pieces,
+        total
+      };
+    },
+    setPieces(data) {
+      this.currentPage = data.currentPage;
+      this.totalPages = data.pages;
+      this.items = data.results;
+      this.filterChoices = data.choices;
+    },
+    async managePieces () {
+      if (this.holdQueries) {
+        return;
+      }
+
+      this.holdQueries = true;
+
+      const {
+        currentPage, pages, results, choices
+      } = await this.requestPieces(this.currentPage);
+      if (!this.isMounted) {
+        return;
+      }
```

Comment:
Is it still needed since this method isn't used in a debounce? 
I don't think we need to worry about what does the state when unmounted, I would be surprised that vue wouldn't handle the case where state is set while component is unmounted.

Reply:
Removed